subprojects/build-operations/src/main/java/org/gradle/trace/buildops/BuildOperationTrace.java

```diff
@@ -37,11 +47,11 @@ public BuildOperationTrace measureConfigurationTime(File outFile) {
     }
 
     public BuildOperationTrace measureBuildOperation(String buildOperationName, File outFile) {
-        Provider<BuildOperationDurationRecordingListener> listenerProvider = sharedServices.registerIfAbsent("build-op-" + buildOperationName, BuildOperationDurationRecordingListener.class, spec -> {
-            spec.getParameters().getCapturedBuildOperation().set(buildOperationName);
-            spec.getParameters().getOutputFile().set(outFile);
+        Provider<BuildOperationDurationRecordingListener> listenerProvider = sharedServices.registerIfAbsent("measure-build-operations", BuildOperationDurationRecordingListener.class, spec -> {
```

Comment:
I don't think registering the same build service twice with different parameters has an effect. We should register the build service only when we know all build operations to register.

This seems to work now since the map is queried each time the `MapProperty` is queried.

Reply:
That is the point, yes.

Reply:
I see how a different design could work, too. Do you feel strongly to change it?

Reply:
I changed the API, so now we have the full map of captured operations up front.